src/_pytest/pathlib.py

```diff
@@ -51,6 +51,8 @@
     1921,  # ERROR_CANT_RESOLVE_FILENAME - fix for broken symlink pointing to itself
 )
 
+TMPDIR_FILE_MODE = int(os.getenv("PYTEST_TMPDIR_FILE_MODE", "0o700"), 8)
```

Comment:
You've chosen to grab the envvar once during import. Other alternatives are to grab it during initialization, and to grab it each time. Any particular reason you chose this way?

Reply:
No particular reason. Preferably I would want to grab it in one place and then share it.

But I'm not versed enough in core to know where and how to do that.

Happy to take some guidance! 🙏

Reply:
IMO (although again, I didn't think it through)

each-time > init > import

init > import because import is not great for this getenv, since then the user doesn't really get a chance to setenv from within python (when using `pytest.main` etc).

each-time > init is more debatable, but each-time allows a user to setenv whenever and it would take effect -- is there any reason to disallow that?

Reply:
> each-time > init is more debatable, but each-time allows a user to setenv whenever and it would take effect -- is there any reason to disallow that?

Not really, I think maximum flexibility is allowed here. I don't think it's our business to protect users from themselves. Especially since the default is more or less the most restrictive.

So would you recommend a small static method somewhere to get the user defined file mode? If so, where should I put such (static) method?

Something like:

```python
def file_mode() -> int:
    return int(os.getenv("PYTEST_TMPDIR_FILE_MODE", "700"), 8)
```

Reply:
Just a note that if we change the default (and remove the `0o`), we should update the docs too. 👍

Reply:
> So would you recommend a small static method somewhere to get the user defined file mode

Yes. Though I'd call it `tmpdir_file_mode`.

> If so, where should I put such (static) method?

I'd say best place for it is `src/_pytest/pathlib.py`.

Reply:
Updated accordingly @bluetech 🙇